tests/helpers/perf_results_helper.rb

```diff
@@ -97,6 +202,210 @@ def csv2html(csv_path)
     # create HTML doc
     heading = "  <h2>#{File.basename(csv_path)}</h2>" + nl
     html = HTML_START + heading + table + HTML_END
-    File.write("#{csv_path}.html", html)
+    html_path = "#{csv_path}.html"
+
+    puts "  creating HTML file: #{html_path}"
+    puts
+
+    File.write(html_path, html)
+  end
+
+  # Extract the HTML table from the csv2html output
+  # so it can be used in template-based release reports
+  #
+  # @author Bill Claytor
+  #
+  # @param [String] html_path The path to the csv2html formatted HTML file
+  #
+  # @return [String] The HTML table
+  #
+  # @example
+  #   results_table = extract_table_from_csv2html_output(html_path)
+  #
+  def extract_table_from_csv2html_output(html_path)
+    puts "extracting table from #{html_path}"
+    puts
+
+    html_string = File.read(html_path)
+    table_string = "".dup
+    table_start = false
+
+    html_string.each_line do |line|
+      table_start = true if line.include?("<table")
+
+      table_string << line if table_start
+
+      break if line.include?("</table>")
+    end
+
+    table_string
+  end
+
+  # Splits the specified atop summary CSV file generated by beaker-benchmark
+  # into separate CSV files for the summary and detail
+  # so they can be converted into HTML tables using csv2html
+  # and used in release reports
+  #
+  # @author Bill Claytor
+  #
+  # @param [String] csv_path The path to the atop csv file
+  #
+  # @return [void]
+  #
+  # @example
+  #   split_atop_csv(csv_path)
+  #
+  # TODO: eliminate the need for this by generating separate csv files after the run
+  #
+  def split_atop_csv_results(csv_path)
+    puts "processing CSV file: #{csv_path}"
+    output_path_summary = csv_path.gsub(".csv", ".summary.csv")
+    output_path_detail = csv_path.gsub(".csv", ".detail.csv")
+
+    output_path = output_path_summary
+
+    contents = File.read(csv_path)
+
+    puts "creating summary: #{output_path_summary}"
+    File.open(output_path_summary, "w")
+
+    puts "creating detail: #{output_path_detail}"
+    File.open(output_path_detail, "w")
+
+    line_ct = 0
+    contents.each_line do |line|
+      line_ct += 1
+
+      output_path = output_path_detail if line_ct > 3
+
+      File.open(output_path, "a") do |f|
```

Comment:
Since the source file only has 11 lines this seemed more straightforward and doesn't risk leaving the file open. Execution is virtually instantaneous :)